src/SFA.DAS.Roatp.CourseManagement.Web/Controllers/AddProviderContact/ConfirmUpdateProviderContactController.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Roatp.CourseManagement.Application.ProviderStandards.Commands.AddProviderContact;
+using SFA.DAS.Roatp.CourseManagement.Web.Infrastructure;
+using SFA.DAS.Roatp.CourseManagement.Web.Infrastructure.Authorization;
+using SFA.DAS.Roatp.CourseManagement.Web.Models.ProviderContact;
+using SFA.DAS.Roatp.CourseManagement.Web.Services;
+
+namespace SFA.DAS.Roatp.CourseManagement.Web.Controllers.AddProviderContact;
+
+[Authorize(Policy = nameof(PolicyNames.HasProviderAccount))]
+[Route("{ukprn}/update-contact-details", Name = RouteNames.AddProviderContact)]
+
+public class ConfirmUpdateProviderContactController(IMediator _mediator, ISessionService _sessionService) : ControllerBase
+{
+    public const string ViewPath = "~/Views/AddProviderContact/UpdateProviderContact.cshtml";
+
+
+    [HttpGet]
+    public IActionResult CheckContact(int ukprn)
+    {
+        var sessionModel = _sessionService.Get<ProviderContactSessionModel>();
+        if (sessionModel == null) return RedirectToRoute(RouteNames.ReviewYourDetails, new { ukprn = Ukprn });
+
+        var model = new ProviderContactUpdateViewModel
+        {
+            EmailAddress = sessionModel.EmailAddress,
+            PhoneNumber = sessionModel.PhoneNumber,
+            ReviewYourDetailsUrl = Url.RouteUrl(RouteNames.ReviewYourDetails, new { ukprn }),
+            ChangeEmailPhoneUrl = Url.RouteUrl(RouteNames.AddProviderContactDetails, new { ukprn }),
+            ShowEmail = !string.IsNullOrEmpty(sessionModel.EmailAddress),
+            ShowPhone = !string.IsNullOrEmpty(sessionModel.PhoneNumber)
```

Comment:
Use IsNullOrWhiteSpace or make sure that when applying the values in the session model we are trimming these down